packages/slate-editor/src/extensions/galleries/components/Gallery/Gallery.tsx

```diff
@@ -162,6 +179,12 @@ export function Gallery({
     }
 
     function handleDelete(id: UniqueIdentifier) {
+        const imageToDelete = images.find((image) => image.id === id);
+        if (!imageToDelete) {
+            return;
+        }
+
+        onImageDelete(imageToDelete);
```

Comment:
My default approach is always to fire tracking/analytics callbacks after the business logic just to avoid analytics errors breaking the main logic.

Maybe we want it here too.

Reply:
Yeah, I see what you mean. But a small problem here is that we're mainly interested in the click events, not when the operation finishes, e.g. caption is edited or image is cropped.

In theory, if I only fire `onImageCropped` at the end of the business logic, people that click "Crop" but then close the Uploadcare dialog will not fire this event, whereas what we want is fire the event for everyone as soon as they click "Crop".

That's why I went with present tense in some cases, otherwise I would just use past tense.

Reply:
> How about onCropClicked? It's past tense, but clear that it's happening before the actual cropping?

Yeah, that works 👍

Reply:
So, do you think it's okay if I just simplify some of the callbacks I've introduced into a respective `onClicked` callback? As we don't really care about the underlying properties or values, we simply care about the click events.

So, it would be:
- `onImageDeleteClicked`,
- `onImageCropClicked`
- `onImageCaptionClicked`

The remaining callbacks can stay in past tense as they are I think.

Reply:
Yeah, sounds good :+1: